### PR TITLE
MAT-4169 UMLS logout

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## Terminology Service PR

Jira Ticket: [MAT-4169](https://jira.cms.gov/browse/MAT-4169)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java [hunk 1]
```diff
@@ -358,4 +358,13 @@ private List<QdmValueSet.Concept> getValueSetConcepts(DescribedValueSet valueSet
   public Optional<UmlsUser> findByHarpId(String harpId) {
     return umlsUserRepository.findByHarpId(harpId);
   }
+
+  public boolean logoutUMLSUser(String userName) {
+    UmlsUser user = verifyUmlsAccess(userName);
+    boolean deleted = false;
+    Optional<UmlsUser> deletedUser = umlsUserRepository.deleteByHarpId(userName);
+    deleted = deletedUser.isPresent();
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/terminology/service/VsacService.java]:
```suggestion
    boolean deleted = deletedUser.isPresent();
```

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java [hunk 2]
```diff
@@ -358,4 +358,13 @@ private List<QdmValueSet.Concept> getValueSetConcepts(DescribedValueSet valueSet
...
+    log.info("Log out UMLS User:{} : {}.", user.getHarpId(), deleted);
```
Comment (RohitKandimalla) [inline src/main/java/gov/cms/madie/terminology/service/VsacService.java]:
```suggestion
    if (deleted) {
          log.info("Successfully deleted UMLS information for User Name: {}", user.getHarpId());
      } else {
          log.error("Error while deleting UMLS information for User Name: {}", user.getHarpId());
      }
```

File: src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java [hunk 3]
```diff
@@ -592,4 +592,34 @@ void testGetCodeStatusIfCodeNotFoundInSvs() {
     CodeStatus status = vsacService.getCodeStatus(code, TEST_API_KEY);
     assertThat(status, is(equalTo(CodeStatus.NA)));
   }
+
+  @Test
+  void testUserUmlsLogout() {
+    when(umlsUserRepository.findByHarpId(anyString())).thenReturn(Optional.of(umlsUser));
+    UmlsUser user = vsacService.verifyUmlsAccess(TEST_API_KEY);
```
Comment (RohitKandimalla) [inline src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java]: I think this is unnecessary 
`vsacService.verifyUmlsAccess(TEST_API_KEY);`

File: src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java [hunk 4]
```diff
@@ -592,4 +592,34 @@ void testGetCodeStatusIfCodeNotFoundInSvs() {
...
+    when(umlsUserRepository.deleteByHarpId(anyString())).thenReturn(Optional.of(umlsUser));
+    boolean loggedOut = vsacService.logoutUMLSUser(umlsUser.getHarpId());
+    assertThat(user.getHarpId(), is(equalTo(TEST_HARP_ID)));
+    assertThat(user.getApiKey(), is(equalTo(TEST_API_KEY)));
+    assertThat(loggedOut, is(equalTo(true)));
```
Comment (RohitKandimalla) [inline src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java]:
```suggestion
    assertTrue(loggedOut);
```

File: src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java [hunk 5]
```diff
@@ -592,4 +592,34 @@ void testGetCodeStatusIfCodeNotFoundInSvs() {
...
+  }
+
```
Comment (RohitKandimalla) [inline src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java:602]: We need another test case to verify when umlsUserRepository.deleteByHarpId(userName) returns Optional.empty() and fails to delete the entry

**Review by RohitKandimalla — CHANGES_REQUESTED** [review]

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java]: @RohitKandimalla done

File: src/main/java/gov/cms/madie/terminology/service/VsacService.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/terminology/service/VsacService.java]: @RohitKandimalla done

File: src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java (hunk 3 above)
Comment (sb-cecilialiu) [reply src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java]: @RohitKandimalla removed

File: src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java (hunk 4 above)
Comment (sb-cecilialiu) [reply src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java]: @RohitKandimalla done

File: src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java (hunk 5 above)
Comment (sb-cecilialiu) [reply src/test/java/gov/cms/madie/terminology/service/VsacServiceTest.java:602]: @RohitKandimalla done

**Review by RohitKandimalla — APPROVED** [review]
